app/main/menu.js

```diff
@@ -211,6 +211,26 @@ class AppMenu {
 				}
 			}, {
 				type: 'separator'
+			}, {
+				label: 'Toggle Do Not Disturb',
+				accelerator: 'Command+Shift+M',
+				click() {
+					const dnd = !ConfigUtil.getConfigItem('dnd', false);
+					const settingList = ['showNotification', 'silent', 'flashTaskbarOnMessage'];
+					let oldSettings;
+					if (dnd) {
+						const currentSettings = {};
+						oldSettings = {};
+						for (const settingName of settingList) {
+							currentSettings[settingName] = ConfigUtil.getConfigItem(settingName);
+							oldSettings[settingName] = (settingName === 'silent');
```

Comment:
This right here seems bit hacky, so i'd add a comment here to explain what's going on for future maintaing.